plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -220,163 +210,51 @@ private static void checkWritable(
+    public static Optional<Boolean> directoryExists(TrinoFileSystem fileSystem, Location path)
     {
         try {
-            return hdfsEnvironment.getFileSystem(context, path).exists(path);
+            return fileSystem.directoryExists(path);
         }
         catch (IOException e) {
             throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed checking path: " + path, e);
         }
     }
 
-    public static boolean isS3FileSystem(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
-    {
-        try {
-            FileSystem fileSystem = getRawFileSystem(hdfsEnvironment.getFileSystem(context, path));
-            return fileSystem instanceof TrinoS3FileSystem || fileSystem.getClass().getName().equals(EMR_FS_CLASS_NAME) || fileSystem instanceof CachingTrinoS3FileSystem;
-        }
-        catch (IOException e) {
-            throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed checking path: " + path, e);
-        }
-    }
-
-    public static boolean isViewFileSystem(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
-    {
-        try {
-            return getRawFileSystem(hdfsEnvironment.getFileSystem(context, path)) instanceof ViewFileSystem;
-        }
-        catch (IOException e) {
-            throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed checking path: " + path, e);
-        }
-    }
-
-    private static boolean isDirectory(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
-    {
-        try {
-            return hdfsEnvironment.getFileSystem(context, path).isDirectory(path);
-        }
-        catch (IOException e) {
-            throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed checking path: " + path, e);
-        }
-    }
-
-    public static boolean isHdfsEncrypted(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path path)
-    {
-        try {
-            FileSystem fileSystem = getRawFileSystem(hdfsEnvironment.getFileSystem(context, path));
-            if (fileSystem instanceof DistributedFileSystem) {
-                return ((DistributedFileSystem) fileSystem).getEZForPath(path) != null;
-            }
-            return false;
-        }
-        catch (IOException e) {
-            throw new TrinoException(HIVE_FILESYSTEM_ERROR, "Failed checking encryption status for path: " + path, e);
-        }
-    }
-
     public static boolean isFileCreatedByQuery(String fileName, String queryId)
     {
         return fileName.startsWith(queryId) || fileName.endsWith(queryId);
     }
 
-    public static Location createTemporaryPath(HdfsContext context, HdfsEnvironment hdfsEnvironment, Path targetPath, String temporaryStagingDirectoryPath)
+    public static Optional<Location> createTemporaryPath(TrinoFileSystem fileSystem, ConnectorIdentity identity, Location targetPath, String temporaryStagingDirectoryPath)
     {
         // use a per-user temporary directory to avoid permission problems
```

Comment:
This comment isn't really accurate.  This is really "interpolate user into path if requested"

Reply:
Why isn't it accurate? That's why we do the interpolation (and `${USER}` is in the default config). I'm guessing this comment predates this being configurable. I can reword the comment.